pkg/k8s/client.go

```diff
@@ -105,11 +108,50 @@ func CreateConfig() (*rest.Config, error) {
 	return createConfig(GetAPIServerURL(), GetKubeconfigPath(), GetQPS(), GetBurst())
 }
 
+func setDialer(config *rest.Config) func() {
+	context := (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}).DialContext
+	dialer := connrotation.NewDialer(context)
+	config.Dial = dialer.DialContext
+	return dialer.CloseAll
+}
+
+func runHeartbeat(closeAllConns []func(), stop chan struct{}) error {
+	timeout := option.Config.K8sHeartbeatTimeout
+	go wait.Until(func() {
+		done := make(chan error)
+		go func() {
+			_, err := k8sCli.Interface.CoreV1().Namespaces().Get("kube-system", metav1.GetOptions{})
+			if err != nil {
+				done <- err
+			}
+
+			close(done)
+		}()
+
+		select {
+		case err := <-done:
+			if err != nil {
```

Comment:
In case the error is type of timeout (504 and / or 408?), we won't execute `closeAllConns`, is this expected?

Reply:
Yeah that makes sense, I changed it such that if there is a status error and either one of those status-codes, it'll restart the connection as well. Perhaps we'd want to add `StatusBadGateway` to that list?

Reply:
Yeah I guess we can add that one as well.